pkg/db/v1alpha2/interface.go

```diff
@@ -95,6 +95,9 @@ func openSQLConn(driverName string, dataSourceName string, interval time.Duratio
 				if err = db.Ping(); err == nil {
 					return db, nil
 				}
+				klog.Infof("Ping to Katib db failed: %v", err)
```

Comment:
Should we return nil, err here?

Reply:
@gaocegege This is in a for-select loop. Shouldn't it wait till timeout?

Reply:
@andreyvelich Just wondering. Have you seen this log https://github.com/kubeflow/katib/blob/master/cmd/manager/v1alpha2/main.go#L199 after the timeout?

Reply:
Oh gotcha, yeah we should wait instead of returning an error.

Reply:
@johnugeorge I didn't check it, because after the timeout my manager container was automatically restarting.